eos-link-host-mali-driver

```diff
@@ -22,14 +22,45 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 EXTENSION_REF=org.freedesktop.Platform.GL.host/arm/1.6
-MALI_DIR=/usr/lib/mali400/arm-linux-gnueabihf
 
-DEST_DIR="/var/lib/flatpak/extension/$EXTENSION_REF"
-mkdir -p $(dirname "$DEST_DIR")
-ln -sf "$MALI_DIR" "$DEST_DIR"
+#
+# create_flatpak_gl_host_extension MALI_DIR FLATPAK_DIR
+#
+# Creates a flatpak GL host extension in $FLATPAK_DIR from $MALI_DIR directory
+# MALI_DIR needs to contain libMali.so and libMaliWrapper.so libraries.
+#
+function create_flatpak_gl_host_extension {
+    MALI_DIR=$1
+    FLATPAK_DIR=$2
+
+    if test ! -d $FLATPAK_DIR; then
+        return;
+    fi
+
+    DEST_DIR=$FLATPAK_DIR/extension/$EXTENSION_REF
+
+    # Ensure destination directory
+    mkdir -p $DEST_DIR
 
-if test -d /var/endless-extra/flatpak; then
-    DEST_DIR="/var/endless-extra/flatpak/extension/$EXTENSION_REF"
-    mkdir -p $(dirname "$DEST_DIR")
-    ln -sf "$MALI_DIR" "$DEST_DIR"
+    # Hardlink mali driver and wrapper
+    ln -t $DEST_DIR $MALI_DIR/libMali.so
+    ln -t $DEST_DIR $MALI_DIR/libMaliWrapper.so
+
+    # Create lib link for flatpak GL extension
+    ln -sr $DEST_DIR $DEST_DIR/lib
+
+    # Create extra EGL and GLES links
+    ln -sr $DEST_DIR/libEGL.so.1.4 $DEST_DIR/libEGL.so.1
+    ln -sr $DEST_DIR/libGLESv2.so.2.0 $DEST_DIR/libGLESv2.so.2
+    ln -sr $DEST_DIR/libMaliWrapper.so $DEST_DIR/libEGL.so.1.4
+    ln -sr $DEST_DIR/libMaliWrapper.so $DEST_DIR/libGLESv2.so.2.0
+}
+
+MALI_DIR="/usr/lib/mali400/arm-linux-gnueabihf"
+
+# Create GL host extensions
+if test -d $MALI_DIR; then
+    create_flatpak_gl_host_extension $MALI_DIR "/var/lib/flatpak"
+    create_flatpak_gl_host_extension $MALI_DIR "/var/endless-extra/flatpak"
```

Comment:
hardlinks don't cross filesystems boundaries, so this would not work for `/var/endless-extra`.

Reply:
Just realized this will not work for `/var/lib/flatpak` either, as due to ostree's bind-mounting `/usr` and `/var` are seen as separate filesystems:
```
$ sudo ln /usr/bin/true /var
ln: failed to create hard link ‘/var/true’ => ‘/usr/bin/true’: Invalid cross-device link
```